custom/icds/utils/ccz_hosting.py

```diff
@@ -0,0 +1,62 @@
+from __future__ import absolute_import
+from __future__ import unicode_literals
+
+from django.urls import reverse
+from django.utils.functional import cached_property
+
+from corehq.blobs import (
+    get_blob_db,
+    CODES,
+)
+from corehq.blobs.models import BlobMeta
+
+
+class CCZHostingUtility:
+    def __init__(self, ccz_hosting=None, blob_id=None):
+        self.ccz_hosting = ccz_hosting
+        self.blob_id = blob_id
+        if ccz_hosting and not blob_id:
+            self.blob_id = self.ccz_hosting.blob_id
+
+    def file_exists(self):
+        return get_blob_db().exists(key=self.blob_id)
+
+    def get_file(self):
+        return get_blob_db().get(key=self.blob_id)
+
+    def get_file_size(self):
+        return get_blob_db().size(key=self.blob_id)
+
+    @cached_property
+    def get_file_meta(self):
```

Comment:
the reason i keep `get_` in the name because the object is getting it or obtaining it from another place and is not set on the object itself and hence there is some db call or call on another object is going to happen. Just keeping it a little explicit.
i think setting it cached_property then looks incorrect and should be just memoized.

Reply:
I agree, it would be better to use `memoized` decorator rather than `cached_property`.